packages/slate-react/src/components/editable.tsx

```diff
@@ -760,11 +760,14 @@ export const Editable = (props: EditableProps) => {
               ) {
                 const node = ReactEditor.toSlateNode(editor, event.target)
                 const path = ReactEditor.findPath(editor, node)
-                if (event.detail === TRIPLE_CLICK) {
+                if (event.detail >= TRIPLE_CLICK && path.length) {
                   const start = Editor.start(editor, [path[0]])
                   const end = Editor.end(editor, [path[0]])
                   const range = Editor.range(editor, start, end)
-                  Transforms.select(editor, range)
+                  setTimeout(() => {
+                    Transforms.select(editor, range)
+                    Editor.unhangRange(editor, range)
+                  }, 100)
```

Comment:
Given the timeout, two potential issues: 1) the range could be invalid by the time the timer expires (editor content could have changed). Calling select() and unhangRange() with an invalid range could cause unexpected behavior including a crash. 2) the component could have unmounted before the timer expires, and these apis should not be called in that case. Making the code synchronous would remove these problems.

#1: I don't know of a fix for this one. It's not clear how you could know the range is valid when the timer fires.
#2: You can check if the component has been unmounted by setting a ref boolean in a useEffect cleanup callback. Then that ref's value can be checked in this setTimeout callback function to see if the component has unmounted. If it has, then don't make these api calls.

Reply:
add `if (!ref.current) return` in setTimeout callback

Reply:
The node, path, and range may not be right after 100ms if editor.children changed within that time. That could be solved by putting all the logic inside the setTimeout callback like this:

```
if (event.detail >= TRIPLE_CLICK) {
  setTimeout(() => {
    if (ref.current) {
      const node = ReactEditor.toSlateNode(editor, event.target)
      const path = ReactEditor.findPath(editor, node)
      if (path.length) {
        const start = Editor.start(editor, [path[0]])
        const end = Editor.end(editor, [path[0]])
        const range = Editor.range(editor, start, end)
        Transforms.select(editor, range)
        Editor.unhangRange(editor, range)
      }
    }
  }, 100);
  return;
}
```

Reply:
putting all the logic inside the setTimeout callback have another problem, the event.target maybe not current target , it 
will set selection to prev target
`onDOMSelectionChange` function in `packages/slate-react/src/components/editable.tsx`
<img width="549" alt="image" src="https://user-images.githubusercontent.com/16861647/163095342-061cf19e-60b6-4ab6-ab2e-75348056d7e0.png">
when i change to `Transforms.unhangRange(editor, range)`, triple and more click is fine, but i don't know  the potential issues

Reply:
@scottlu i delete the setTimeout function, and use ` state.isTripleClick ` to check 3 or more clicks, on `onDOMSelectionChange` event to decide how to set the selection, what do you think?